### ISSUE TITLE
Show_session recalculates session value 

### ISSUE THREAD

**Issue by InnaXITE** [agourlay/cornichon#177]
Hi, I was writing a test case with the next steps:

```scala
Given I save("deviceId" → "<random-uuid>")
And I show_session("deviceId")
```

So upon execution I'm getting this:

```
Given I add value '7f93126d-4f03-431e-b872-a067e92614fa' to session under key 'deviceId'  (18 millis)
Session content for key 'deviceId' is 943faa48-5262-4875-a51a-5df514447aba
```

Then session value persists across the tests with the value#2 (from debug message).

I suppose it might be related to Cats framework but nor sure. For my scenario it looks rather as a defect.

Thank you!

**Comment by agourlay** [comment]
Hi @InnaXITE,

thank you for the bug report :)

I have first a quick question, are using the `random-uuid` placeholder in your scenario?
`Given I save("deviceId" -> "<random-uuid>")`

or are you writing the UUID yourself?

**Comment by InnaXITE** [comment]
Hi Arnaud!

Ah, thank you for getting back to me so fast! This is my first bug report to open source product hehe

Yes, I’m using random one.

If I can help you more with the details - let me know!

Enjoy this day,
Inna



Inna Shevchenko
Scrum Master & QA Engineer

XITE Networks International
Spijkerkade 28
1021 JS Amsterdam

T:	+31(0)20 715 52 07
M:	+31 (6)40 89 20 24 
W:	xite.com

**Comment by agourlay** [comment]
Then I think I know what the problem is.

When you write `Given I save("deviceId" -> "<random-uuid>")` the placeholder is actually resolved twice.

Once for generating the title of the steps in the log and once for actually performing the operation itself.

Usually this is not a problem, because placeholder resolution always yields the same result for a given step...

Except when using a non-deterministic placeholder such as `<random-uuid>`.

Now the good news is that the value in `Session` is correct.
That is the `show_session` is actually showing you the right value.

The bad news is that the value displayed in the log is incorrect.

In practice, this is not such a big deal because the test does not rely on the value in the logs.

I do not have a fix for this problem yet because it is not so easy to solve.

**Comment by agourlay** [comment]
FYI in `0.19.0` the random placeholders won't be resolved in the title anymore.
This will remove the confusion.

**Comment by agourlay** [comment]
fixed in `0.19.0`
